include/yaml-cpp/node/detail/impl.h

```diff
@@ -58,19 +58,24 @@ struct get_idx<Key, typename std::enable_if<std::is_signed<Key>::value>::type> {
 
 template <typename Key, typename Enable = void>
 struct remove_idx {
-  static bool remove(std::vector<node*>&, const Key&) { return false; }
+  static bool remove(std::vector<node*>&, const Key&, std::size_t*) {
+    return false;
+  }
 };
 
 template <typename Key>
 struct remove_idx<
     Key, typename std::enable_if<std::is_unsigned<Key>::value &&
                                  !std::is_same<Key, bool>::value>::type> {
 
-  static bool remove(std::vector<node*>& sequence, const Key& key) {
+  static bool remove(std::vector<node*>& sequence, const Key& key,
+                     std::size_t* seqSize) {
     if (key >= sequence.size()) {
       return false;
     } else {
+      bool defined = sequence[key]->is_defined();
       sequence.erase(sequence.begin() + key);
+      *seqSize -= static_cast<size_t>(defined && *seqSize > key);
```

Comment:
Prefer this as an "if" statement.

Reply:
Assuming `seqSize` will be a `std::size_t&` instead of a `std::size_t*`, this should make it clear:
```
      sequence.erase(std::next(sequence.begin(), key));
      seqSize = std::distance(sequence.begin(), sequence.end());
```

Reply:
I've changed it to an if-statement, and also removed the condition of it being defined. I realized that if seqSize was already greater than the key/index, then the node at one point must have been defined and it's only a pathological case where someone specifically set the Node = undefined.